<launch>
  <node type="hokuyo_node" pkg="hokuyo_node" name="hokuyo_node_move">
    <!--frame_idはURGが載ってるフレームに合わせる-->
    <param name="frame_id" value="laser_link" />
    <param name="port" value="/dev/ttyACM0" />
    <param name="calibrate_time" value="false" />
    <param name="skip" value="true" />
    <param name="intensity" value="true" />
    <param name="min_ang" value="-1.047" />
    <param name="max_ang" value="1.047" />
    <remap from="scan" to="move_scan" />
  </node>
  <!--サーボを周期的に動かしてtfしてるだけ-->
  <node type="futaba_driver" pkg="futaba_driver" name="futaba_driver">
    <param name="port_name" value="/dev/ttyUSB0" />
  </node>
  <!--laser_assemblerのsrvを定期的に呼び出してPointCloudをPublishしてる-->
  <node type="periodic_snapshotter" pkg="laser_assembler" name="my_laser_assembler">
  </node>
  <!--laserをassembleしてる-->
  <!--fixed_frameは固定するフレームを指定する-->
  <node type="laser_scan_assembler" pkg="laser_assembler" name="synchronize_assembler">
    <remap from="scan" to="move_scan" />
    <param name="max_scans" type="int" value="1000" />
    <param name="fixed_frame" type="string" value="map" />
  </node>
</launch>
